### ISSUE TITLE
Pressin "Next" at step 1 of upload does nothing

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#2376]
![screenshot_20190130-224319](https://user-images.githubusercontent.com/99590/51985264-d701bb80-24e0-11e9-8879-bcd68dfde02b.png)

**Comment by vanshikaarora** [comment]
Hey @nicolas-raoul Is this Issue locale specific? 
I cannot reproduce this issue, On pressing next with empty description It takes me to the step 2

**Comment by maskaravivek** [comment]
AFAIK it should not happen consistently. It might happen if some of the image preprocessing is still pending. 

@nicolas-raoul I made the following changes in the preprocessing step: 
- made it async
- moved it to `computation` thread from `main` thread

I didn't realize that it could cause this issue. How do you suggest to address it? I can think of 2 solutions: 
- make the preprocessing step blocking. 
- show a loader on clicking next, if the preprocessing for that image is still not complete. 

Preprocessing has: 
- An API call to check for duplicate name [depends on network strength + image title(it will keep making API calls until it gets a unique name)]
- check for dark image [can be time consuming based on image size]
- check for image location in case of nearby upload [should be quick]

**Comment by nicolas-raoul** [comment]
I happened to me yesterday morning in the Chinese locale. I switched to English and then it worked, but that might be caused by other things, maybe the app was kicked out of memory while I was switching languages or something.
Then at noon I uploaded a few pictures successfully.
Then in the evening after updating I faced the same problem, again with Chinese. I switch to German, same problem, then to English, same problem.
